File: backend/src/main/java/com/github/ingbeck/backend/model/meal/Meal.java
package com.github.ingbeck.backend.model.meal;


import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document("meals")
public record Meal(
        String id,
        String userId,
        String name,
        int totalCalories,
        List<MealItem> mealItems
) {
}

File: frontend/src/types/Meal.ts
export type Meal = {
    id: string,
    userId: string,
    name: string,
    totalCalories : number,
    mealItems: MealItem[]
}

export type MealToSaveDto = {
    name: string,
    mealItems: MealItem[]
}

export type MealItem = {
    id: string,
    name: string,
    amount: number,
    unit: string,
    calories: number,
    energyKcal100: number
}
